### PR TITLE
Subresource definition ADR proposal

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | na
| New feature?  | na
| BC breaks?    | no
| Deprecations? | na
| Tickets       |  #2706
| License       | MIT
| Doc PR        | na

@GregoireHebert @dunglas @vincentchalamon @Deuchnord

### CODE REVIEW

File: docs/adr/0000-subresources-definition.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+* The `ApiSubresource` annotation is declared on a Model's properties, which was identified as the root of several issues. For example, finding what class it is defined on ([#3458][issue/3458]). Having multiple `ApiResource` would improve a lot the declaration of our internal metadata and would cause less confusion for developers. 
+* The `path` of these multiple `ApiResource` needs to be implicitly described. 
+
+### Examples
+
+A Company resource with a Company Users subresource on (`/companies/1/users`);
+
+```php
+/**
+ * @ApiResource()
```
Comment (vincentchalamon) [inline docs/adr/0000-subresources-definition.md:30]: > The `path` of these multiple `ApiResource` needs to be implicitly described.
```suggestion
 * @ApiResource(path="/companies")
```

File: docs/adr/0000-subresources-definition.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * @ApiResource(path="/companies/{companyId}/users")
```
Comment (vincentchalamon) [inline docs/adr/0000-subresources-definition.md:31]: Isn't it supposed to be on the User entity instead?

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

**Review by vincentchalamon — APPROVED** [review]

File: docs/adr/0000-subresources-definition.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+* Status: proposed
+* Deciders: @dunglas, @vincentchalamon, @soyuka, @GregoireHebert, @Deuchnord
+
+## Context and Problem Statement
+
+Subresources introduced in 2017 ([#904][pull/904]) the `ApiSubresource` annotation. This definition came along with its own set of issues ([#2706][issue/2706]) and needs a refreshment. On top of that, write support on subresources is a wanted feature and it is hard to implement currently ([#2598][pull/2598]) (See [ADR-0001-subresource-write-support](./0001-subresource-write-support.md)). How can we revamp the Subresource definition to improve the developer experience and reduce the complexity?
+
+## Considered Options
+
+* Fix the actual `ApiSubresource` annotation
```
Comment (Deuchnord) [inline docs/adr/0000-subresources-definition.md:12]:
```suggestion
* Fix the current `ApiSubresource` annotation
```

**Review by Deuchnord — APPROVED** [review]

File: docs/adr/0000-subresources-definition.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
+# Subresource definition
```
Comment (dunglas) [inline docs/adr/0000-subresources-definition.md:1]:
```suggestion
# Subresource Definition
```

**Review by dunglas — APPROVED** [review]

**Pull request merged**